lms/djangoapps/mobile_api/video_outlines/serializers.py

```diff
@@ -159,23 +159,33 @@ def find_urls(course_id, block, child_to_parent, request):
     return unit_url, section_url
 
 
-def video_summary(course, course_id, video_descriptor, request, local_cache):
+def video_summary(video_profiles, course_id, video_descriptor, request, local_cache):
     """
     returns summary dict for the given video module
     """
-    # First try to check VAL for the URLs we want.
-    val_video_info = local_cache['course_videos'].get(video_descriptor.edx_video_id, {})
-    if val_video_info:
-        video_url = val_video_info['url']
+    # Get encoded videos
+    encoded_videos = local_cache['course_videos'].get(video_descriptor.edx_video_id, {})
+
+    # Get highest priority video to populate backwards compatible field
+    default_encoded_video = {}
+
+    if encoded_videos:
+        for profile in video_profiles:
+            if encoded_videos['profiles'].get(profile):
+                default_encoded_video = encoded_videos['profiles'].get(profile)
+                break
+
+    if default_encoded_video:
+        video_url = default_encoded_video['url']
     # Then fall back to VideoDescriptor fields for video URLs
     elif video_descriptor.html5_sources:
         video_url = video_descriptor.html5_sources[0]
     else:
         video_url = video_descriptor.source
 
-    # If we have the video information from VAL, we also have duration and size.
-    duration = val_video_info.get('duration', None)
-    size = val_video_info.get('file_size', 0)
+    # Get duration/size, else default
+    duration = encoded_videos.get('duration', None)
```

Comment:
I don't think it should be popped. Rather, the code below should extract the profiles field.